src/lando/main/scm/git.py

```diff
@@ -0,0 +1,258 @@
+
+        Returns:
+            None
+        """
+        command = ["push"]
+        if force_push:
+            command += ["--force"]
+        command += [push_path]
+        if target:
+            command += [f"HEAD:{target}"]
+        self._git_run(*command, cwd=self.path)
+
+    def last_commit_for_path(self, repo_path: str, path: str) -> str:
+        """Find last commit to touch a path.
+
+        Args:
+            repo_path (str): The path to the repository.
+            path (str): The specific path within the repository.
+
+        Returns:
+            str: The commit id
+        """
+        command = ["log", "--max-count=1", "--format=%H", "--", path]
+        return self._git_run(*command, cwd=repo_path)
+
+    def apply_patch(
+        self, diff: str, commit_description: str, commit_author: str, commit_date: str
+    ):
+        """Apply the given patch to the current repository
+
+        Args:
+            patch_buf (StringIO): The patch to apply
+
+        Returns:
+            None
+        """
+        f_msg = tempfile.NamedTemporaryFile(encoding="utf-8", mode="w+")
+        f_diff = tempfile.NamedTemporaryFile(encoding="utf-8", mode="w+")
+        with f_msg, f_diff:
+            f_msg.write(commit_description)
+            f_msg.flush()
+            f_diff.write(diff)
+            f_diff.flush()
+
+            cmds = [
+                ["apply", f_diff.name],
+                ["add", "-A"],
+                [
+                    "commit",
+                    "--date",
+                    commit_date,
+                    "--author",
+                    commit_author,
+                    "--file",
+                    f_msg.name,
+                ],
+            ]
+
+            for c in cmds:
+                self._git_run(c, cwd=self.path)
+
+    @contextmanager
+    def for_pull(self) -> ContextManager:
+        """Context manager to prepare the repo with the correct environment variables set for pulling."""
+        yield self
+
+    @contextmanager
+    def for_push(self, requester_email: str) -> ContextManager:
+        """Context manager to prepare the repo with the correct environment variables set for pushing.
+
+        Args:
+            requester_email (str)
+        """
+        # We set the committer name to the requester's _email_ as this is the only piece
+        # of information about the user that we are comfortable making public. Names in
+        # the User objects are coming from LDAP, and may not be acceptable to use
+        # publicly.
+        os.environ[ENV_COMMITTER_NAME] = requester_email
+        os.environ[ENV_COMMITTER_EMAIL] = requester_email
+        logger.debug(
+            f"{ENV_COMMITTER_EMAIL} and {ENV_COMMITTER_NAME} set to {requester_email}"
+        )
+        try:
+            yield self
+        finally:
+            del os.environ[ENV_COMMITTER_NAME]
+            del os.environ[ENV_COMMITTER_EMAIL]
```

Comment:
I think it may be a little better to forego env vars here and instead use command line arguments in the respective commands? (e.g., `--author`)

Reply:
Looking through the manpage, it doesn't look like there is a `--committer` equivalent to `--author`. 

We may be able to do this via the `git config`. That would a bit too persistent for my taste I'd rather it be explicit in our code, rather than magically set in the persistent config of a dependency.

What do you think?